### PR TITLE
haxor-news: use python3, fix build

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
related: #101964

also fixes build

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
cc @matthiasbeyer

File: pkgs/applications/misc/haxor-news/default.nix [hunk 1]
```diff
@@ -1,20 +1,33 @@
-{ stdenv, python, fetchpatch }:
-
-with python.pkgs;
+{ stdenv, fetchFromGitHub, python3, fetchpatch }:
+
+
+let
+  py = python3.override {
```
Comment (FRidh) [inline pkgs/applications/misc/haxor-news/default.nix:5]: Don't forget the `self = py;`. Won't matter in this case though.

File: pkgs/applications/misc/haxor-news/default.nix [hunk 2]
```diff
@@ -1,20 +1,35 @@
...
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "5b9af8338a0f8b95a8133b66ef106553823813ac171c0aefa3f3f2dbeb4d7f88";
-  };
-
-  # allow newer click version
-  patches = fetchpatch {
-    url = "${meta.homepage}/commit/5b0d3ef1775756ca15b6d83fba1fb751846b5427.patch";
-    sha256 = "1551knh2f7yarqzcpip16ijmbx8kzdna8cihxlxx49ww55f5sg67";
+  version = "unstable-2020-10-20";
```
Comment (matthiasbeyer) [inline pkgs/applications/misc/haxor-news/default.nix:25]: I'm not sure here... I mean, does it hurt to keep the old release and have an `haxor-news-unstable` attribute?

I also think that this piece of software is nothing critical, so we _could_ do it this way, but I'd prefer another maintainer if we do it. You're welcome to add yourself (in a dedicated commit I guess) to the maintainers!

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/haxor-news/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/applications/misc/haxor-news/default.nix:25]: if we do this, the old version will be marked vulnerable very soon, see #101964

**Review by matthiasbeyer — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102391` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>haxor-news</li>
  </ul>
</details>

**Comment by erictapen** [comment]
@jonringer Sure this is backport worthy? Builds for me on 20.09 and it doesn't seem to contain any security fixes?

**Pull request merged**